Actions/WorkflowPostProcess/WorkflowPostProcess.ps1

```diff
@@ -1,26 +1,114 @@
+        if ($null -ne $timedOutJobs) {
+            return "TimedOut"
+        }
+        $cancelledJobs = $workflowJobs.jobs | Where-Object { $_.conclusion -eq "cancelled" }
+        if ($null -ne $cancelledJobs) {
+            return "Cancelled"
+        }
+    }
 
-    Write-Host "Post-processing workflow $eventId"
-    if ($telemetryScopeJson -and $telemetryScopeJson -ne '7b7d') {
-        $telemetryScope = RegisterTelemetryScope (hexStrToStr -hexStr $telemetryScopeJson)
-        TrackTrace -telemetryScope $telemetryScope
+    return "Success"
+}
+
+function GetAlGoVersion($ActionRef) {
+    if ($ENV:GITHUB_REPOSITORY -eq "microsoft/AL-Go") {
+        return "Preview"
+    } elseif($ENV:GITHUB_REPOSITORY -notlike "microsoft/*") {
+        return "Developer/Private"
+    } else {
+        return $ActionRef
     }
 }
-catch {
-    if (Get-Module BcContainerHelper) {
-        TrackException -telemetryScope $telemetryScope -errorRecord $_
+
+function LogWorkflowEnd($TelemetryScopeJson, $JobContext, $AlGoVersion) {
+    [System.Collections.Generic.Dictionary[[System.String], [System.String]]] $AdditionalData = @{}
+    $telemetryScope = $null
+    if ($TelemetryScopeJson -ne '') {
+        $telemetryScope = $TelemetryScopeJson | ConvertFrom-Json
+    }
+
+    # Get the workflow conclusion
+    $workflowConclusion = GetWorkflowConclusion -JobContext $JobContext
+    Add-TelemetryProperty -Hashtable $AdditionalData -Key 'WorkflowConclusion' -Value $workflowConclusion
+
+    # Calculate the workflow duration using the github api
+    if ($telemetryScope -and ($null -ne $telemetryScope.workflowStartTime)) {
+        Write-Host "Calculating workflow duration..."
+        $workflowTiming= [DateTime]::UtcNow.Subtract([DateTime]::Parse($telemetryScope.workflowStartTime)).TotalSeconds
+        Add-TelemetryProperty -Hashtable $AdditionalData -Key 'WorkflowDuration' -Value $workflowTiming
+    }
+
+    # Log additional telemetry from AL-Go settings
+    $alGoSettingsPath = "$ENV:GITHUB_WORKSPACE/.github/AL-Go-Settings.json"
+    if (Test-Path -Path $alGoSettingsPath) {
+        $repoSettings = Get-Content -Path $alGoSettingsPath -Raw -Encoding UTF8 | ConvertFrom-Json
+
+        if ($repoSettings.PSObject.Properties.Name -contains 'type') {
+            Add-TelemetryProperty -Hashtable $AdditionalData -Key 'RepoType' -Value $repoSettings.type
+        }
+
+        if ($repoSettings.PSObject.Properties.Name -contains 'githubRunner') {
+            Add-TelemetryProperty -Hashtable $AdditionalData -Key 'GitHubRunner' -Value $repoSettings.githubRunner
+        }
+
+        if ($repoSettings.PSObject.Properties.Name -contains 'runs-on') {
+            Add-TelemetryProperty -Hashtable $AdditionalData -Key 'RunsOn' -Value $repoSettings.'runs-on'
+        }
     }
-    throw
+
+    if ($AlGoVersion -ne '') {
+        Add-TelemetryProperty -Hashtable $AdditionalData -Key 'ALGoVersion' -Value $AlGoVersion
+    }
+
+    if ($workflowConclusion -eq "Failure") {
```

Comment:
Just a suggestion: instead of all the cases, how about just "AL-Go workflow finished/completed with status $workflowConclusion..."

Reply:
The cases are also there because Trace-Information and Trace-Exception log with different severity level to app insights. That being said, we can probably group "Failure" and "Timedout" into "AL-Go workflow failed" while letting the rest be logged as "AL-Go workflow ran". 
Regardless of which message we log the telemetry with, we'll still include the WorkflowConclusion dimension in the telemetry so you can still query for TimedOut workflows simply by filtering on this dimension.